portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,80 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    Query query = new Query("Comment");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> messages = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("content");
+
+      messages.add(comment);
+    }
+
+    String json = convertToJson(messages);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts an instance into a JSON string using manual String concatentation.
+   */
+  private String convertToJson(ArrayList<String> messages) {
+    String json = "[";
+    for(int i = 0; i < messages.size(); ++i){
+        if(i != 0)
+            json += ", ";
+        json += '{';
+        json += "\"comment"+ "\": ";
+        json += "\"" + messages.get(i) + "\"";
+        json += '}';
+    }
+    json += "]";
+    return json;
+  }
```

Comment:
a suggestion: use StringBuilder

https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html

Reply:
Or use Stream in Java 8 style. For example:

String commaSeapratedMessage = messages.stream()
  .map(msg -> String.format("{\"comment\": \"%s\"}", msg))
  .collect(Collectors.joining(", "));
return "[" + commaSeparatedMessage + "]";

Reply:
Unresolve the comment

Reply:
Updated my code, using Stream in Java 8 style is indeed more convenient and clearer.